newsroom/monitoring/views.py

```diff
@@ -188,86 +201,111 @@ async def edit(_id):
-    return jsonify(profile), 200
+            await MonitoringProfileService().update(args.item_id, updates)
+            return Response({"success": True})
+        return Response(form.errors, 400)
+    return Response(profile)
 
 
-@blueprint.route("/monitoring/<_id>", methods=["DELETE"])
-@admin_only
-async def delete(_id):
+@monitoring_endpoints.endpoint("/monitoring/<_id>", methods=["DELETE"], auth=[auth_rules.admin_only])
+async def delete(args: MonitoringIdUrlArg, params: None, request: Request) -> Response:
     """Deletes the monitoring profile by given id"""
-    get_resource_service("monitoring").delete_action({"_id": ObjectId(_id)})
-    return jsonify({"success": True}), 200
+    service = MonitoringProfileService()
+    monitoring = await service.find_by_id(args.id)
+    if not monitoring:
+        return Response({"error": gettext("Item not found")}, 404)
+    await MonitoringProfileService().delete(monitoring)
+    return Response({"success": True})
 
 
-@blueprint.route("/monitoring")
-@section("monitoring")
-@login_required
+@monitoring_endpoints.endpoint("/monitoring", auth=[auth_rules.section_required("monitoring")])
 async def index():
     data = await get_view_data()
     user_profile_data = await get_user_profile_data()
     return await render_template("monitoring_index.html", data=data, user_profile_data=user_profile_data)
 
 
-@blueprint.route("/monitoring/export/<_ids>")
-@login_required
-async def export(_ids):
-    user = get_user_from_request(None)
-    _format = request.args.get("format")
-    if not _format:
-        return jsonify({"message": "No format specified."}), 400
+class ExportMonitoringUrlArgs(BaseModel):
+    ids: list[str]
+
+    @field_validator("ids", mode="before")
+    def parse_ids(cls, value: list[str] | str) -> list[str]:
+        """If value is not a list, then convert it to a list here"""
+
+        return value.split(",") if isinstance(value, str) else value
 
-    layout_format = request.args.get("secondary_format")
-    formatter = get_current_app().as_any().download_formatters[_format]["formatter"]
-    monitoring_profile = get_entity_or_404(request.args.get("monitoring_profile"), "monitoring")
-    monitoring_profile["format_type"] = _format
-    monitoring_profile["alert_type"] = layout_format
-    items = get_items_for_monitoring_report([_id for _id in _ids.split(",")], monitoring_profile)
+
+class ExportMonitoringUrlParams(BaseModel):
+    monitoring_profile: ObjectId
+    format: str | None = None
+    secondary_format: str | None = None
+
+
+@monitoring_endpoints.endpoint("/monitoring/export/<_ids>")
+async def export(args: ExportMonitoringUrlArgs, params: ExportMonitoringUrlParams, request: Request) -> Response:
+    user = get_user_from_request(request)
+    if not params.format:
+        return Response({"message": gettext("No format specified")}, 400)
+
+    formatter = get_current_app().as_any().download_formatters[params.format]["formatter"]
```

Comment:
if you use `newsroom.core.get_current_wsgi_app` you won't need the `as_any`

```suggestion
    formatter = get_current_wsgi_app().download_formatters[params.format]["formatter"]
```